contracts/FlowStakingCollection.cdc

```diff
@@ -0,0 +1,844 @@
+                    self.unlockedTokensUsed = self.unlockedTokensUsed - from.balance
+
+                    self.unlockedVault.borrow()!.deposit(from: <-from)
+                } else {
+                    // Return unlocked tokens first
+                    self.unlockedVault.borrow()!.deposit(from: <-from.withdraw(amount: self.unlockedTokensUsed))
+                    self.unlockedTokensUsed = 0.0
+
+                    self.lockedTokensUsed = self.lockedTokensUsed - from.balance
+                    // followed by returning the difference as locked tokens
+                    self.lockedVault!.borrow()!.deposit(from: <-from)
+                }
+            } else {
+                self.unlockedTokensUsed = self.unlockedTokensUsed - from.balance
+                
+                // If there is no locked account, get the users vault capability and deposit tokens to it.
+                self.unlockedVault.borrow()!.deposit(from: <-from)
+            }
+        }
+
+        /// Returns true if a Stake or Delegation record exists in the StakingCollection for a given nodeID and optional delegatorID, otherwise false.
+        pub fun doesStakeExist(nodeID: String, delegatorID: UInt32?): Bool {
+            var tokenHolderNodeID: String? = nil
+            var tokenHolderDelegatorNodeID: String? = nil
+            var tokenHolderDelegatorID: UInt32?  = nil
+
+            // If there is a locked account, get the staking info from that account
+            if self.tokenHolder != nil {
+                if let _tokenHolder = self.tokenHolder!.borrow() {
+                    tokenHolderNodeID = _tokenHolder!.getNodeID()
+                    tokenHolderDelegatorNodeID = _tokenHolder!.getDelegatorNodeID()
+                    tokenHolderDelegatorID = _tokenHolder!.getDelegatorID()
+                }
+            }
+
+            // If the request is for a delegator, check all possible delegators for possible matches
+            if let _delegatorID = delegatorID {
+                if (tokenHolderDelegatorNodeID != nil && tokenHolderDelegatorID != nil && tokenHolderDelegatorNodeID! == nodeID && tokenHolderDelegatorID! == _delegatorID) {
+                    return true
+                }
+
+                // Look for a delegator with the specified node ID and delegator ID
+                return self.borrowDelegator(nodeID, _delegatorID) != nil 
+            } else {
+                if (tokenHolderNodeID != nil && tokenHolderNodeID! == nodeID) {
+                    return true
+                }
+
+                return self.borrowNode(nodeID) != nil
+            }
+        }
+
+        /// Function to add an existing NodeStaker object
+        pub fun addNodeObject(_ node: @FlowIDTableStaking.NodeStaker) {
+            let stakingInfo = FlowIDTableStaking.NodeInfo(nodeID: node.id)
+            let totalStaked = stakingInfo.tokensStaked + stakingInfo.tokensCommitted + stakingInfo.tokensUnstaking + stakingInfo.tokensUnstaked
```

Comment:
This summing (`tokensStaked + tokensCommitted + tokensUnstaking + tokensUnstaked`) is repeated many times in the contract. Consider adding a function and using it here and in the other places